client/coral-admin/src/components/UserDetail.js

```diff
@@ -100,36 +87,90 @@ export default class UserDetail extends React.Component {
     );
   }
 
+  getActionMenuLabel() {
+    const {root: {user}} = this.props;
+
+    if (isBanned(user)) {
+      return 'Banned';
+    } else if (isSuspended(user)) {
+      return 'Suspended';
+    }
+
+    return '';
+  }
+
   renderLoaded() {
     const {
       data,
       root,
       root: {
+        me,
         user,
         totalComments,
         rejectedComments,
-        comments: {nodes, hasNextPage}
+        comments: {
+          nodes,
+        }
       },
       activeTab,
       selectedCommentIds,
       toggleSelect,
       hideUserDetail,
       viewUserDetail,
       loadMore,
-      toggleSelectAll
+      toggleSelectAll,
+      unBanUser,
+      unSuspendUser,
     } = this.props;
 
     let rejectedPercent = (rejectedComments / totalComments) * 100;
-    if (rejectedPercent === Infinity || isNaN(rejectedPercent)) {
 
-      // if totalComments is 0, you're dividing by zero, which is naughty
+    if (rejectedPercent === Infinity || isNaN(rejectedPercent)) {
       rejectedPercent = 0;
     }
 
+    const banned = isBanned(user);
+    const suspended = isSuspended(user);
+    
     return (
       <ClickOutside onClickOutside={hideUserDetail}>
```

Comment:
Need some logic hear to only hide when no dialog is shown

Reply:
Which dialog? The drawer?

Reply:
I'd add a new reducer called `ui` with one property `modal: Boolean` and pass that to this component e.g. `onClickOutside={!modal && hideUserDetail}`

Reply:
`modal` should be true whenever a dialog is shown (BanDialog, SuspendDialog). The reducer can just act upon the actions from BanDialog and SuspendDialog

Reply:
and false when they dissappear

Reply:
Done :D